src/audio/module_adapter/module_adapter.c

```diff
@@ -138,10 +139,13 @@ static void module_adapter_calculate_dp_period(struct comp_dev *dev)
 	unsigned int period = UINT32_MAX;
 
 	for (int i = 0; i < mod->num_of_sinks; i++) {
-		/* calculate time required the module to provide OBS data portion - a period */
-		unsigned int sink_period = 1000000 * sink_get_min_free_space(mod->sinks[i]) /
-					   (sink_get_frame_bytes(mod->sinks[i]) *
-					   sink_get_rate(mod->sinks[i]));
+		/* calculate time required the module to provide OBS data portion - a period
+		 * use 64bit integers to avoid overflows
+		 */
+		unsigned int sink_period =
+			(uint64_t)1000000 * (uint64_t)sink_get_min_free_space(mod->sinks[i]) /
+			((uint64_t)sink_get_frame_bytes(mod->sinks[i]) *
+			(uint64_t)sink_get_rate(mod->sinks[i]));
```

Comment:
Nit: In multiplication it's enough to type cast the first operand to int64_t to calculate the product as 64 bit. It would give a bit more compact code.